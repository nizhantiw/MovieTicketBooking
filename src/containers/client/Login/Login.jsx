import React from 'react';
import { Link } from 'react-router-dom';


const Login = () => {
  const speak = msg => {
    const sp = new SpeechSynthesisUtterance(msg);
    [sp.voice] = speechSynthesis.getVoices();
    speechSynthesis.speak(sp);
  };
  speak('Hello There, ' + 'You are welcome to Movie Booking Tickets! ' + 'Please, go ahead and signup. Login instead, if you already have an account.');
  
  return (
    <div style={{backgroundColor: "lightcyan", height: "100vh", textAlign: "center"}} >
      <h3>Would you like to SignIn or SignUp?</h3> <br/> <br/> <br/><br/>
      <div>
      <Link to="/login/login"><h1>Login</h1></Link>  <br/> <br/><br/>
      <Link to="/login/register"><h1>Register</h1></Link>   
      </div>
    </div>
  );
};

export default Login;
