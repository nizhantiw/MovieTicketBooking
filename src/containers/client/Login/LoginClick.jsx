import React, { useState } from "react";

const LoginClick = () => {
  const [paymentData, setpaymentData] = useState({
    bankName: "",
    IFSC: "",
  });

  const updatepaymentData = event =>
    setpaymentData({
      ...paymentData,
      [event.target.name]: event.target.value
    });

    const handleFormSubmit=(event)=> {
        alert("Login Successful!!");
        event.preventDefault();
      }

  const { bankName, IFSC } = paymentData;

  return (<div style={{backgroundColor: "lightcyan", height: "100vh",  textAlign:"center"}} >
    <br/><h3>Login with your credentials</h3><br/>
    {/*<img src={signIn} style={{height:"570px"}} />*/}
    <form class="submit" onSubmit={e => handleFormSubmit(e)}>
     <div>
     <label htmlFor="UserId">UserId</label> <br/>
      <input id="UserId"
        value={bankName}
        onChange={e => updatepaymentData(e)}
        placeholder="UserId"
        type="text"
        name="bankName"
        required
      /></div><br/>
    
      <div>
      <label htmlFor="Password">Password</label> <br/>
      <input id="Password"
        value={IFSC}
        onChange={e => updatepaymentData(e)}
        placeholder="Password"
        type="IFSC"
        name="IFSC"
        required
      /></div><br/>
      <input type="submit" value="Submit" />
    </form>
    </div>
  );
};

export default LoginClick;
