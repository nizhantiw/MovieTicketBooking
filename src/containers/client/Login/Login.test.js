import Login from './Login';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { render, fireEvent } from "@testing-library/react";

configure({adapter: new Adapter()});

it("renders correctly", () => {
  const wrapper = shallow(
    <Login />
  );
  expect(wrapper).toMatchSnapshot();
});

/*it('renders a snapshot', () => {
    const tree = renderer.create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
  });*/
  

  it('should show the text', () => {
    const toggleInstance = shallow(<Login />);
    const element = toggleInstance.find('div h3');
    expect(element.text()).toMatchSnapshot();
   });

   it('renders results after search', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper.find('h3').length).toEqual(1);
  });
  it('starts with a count of 0', () => {
      const wrapper = shallow(<Login />);
      const text = wrapper.find('h3').text();
      expect(text).toEqual('Would you like to SignIn or SignUp?');
    });