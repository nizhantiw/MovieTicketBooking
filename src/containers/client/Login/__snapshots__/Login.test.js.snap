// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`should show the text 1`] = `"Would you like to SignIn or SignUp?"`;
